### PR TITLE
Enforce model checkpoints existing for endpoint/bundle creation

### PR DESCRIPTION
**Pull request by dmchoiboi**
# Pull Request Summary

_What is this PR changing? Why is this change being made? Any caveats you'd like to highlight? Link any relevant documents, links, or screenshots here if applicable._

Enforce that model checkpoints are used when creating bundles to prevent outages/issues in prod where a hf repo becomes unavailable

## Test Plan and Usage Guide

_How did you validate that your PR works correctly? How do you run or demo the code? Provide enough detail so a reviewer can reasonably reproduce the testing procedure. Paste example command line invocations if applicable._

### CODE REVIEW

File: model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py [hunk 1]
```diff
@@ -449,22 +449,26 @@ async def create_text_generation_inference_bundle(
...
-                    final_weights_folder,
-                )
-            else:
-                raise ObjectHasInvalidValueException(
-                    f"Only S3 paths are supported. Given checkpoint path: {checkpoint_path}."
-                )
+        if not checkpoint_path:
+            raise InvalidRequestException(f"No checkpoint path found for model {model_name}")
+
+        if checkpoint_path.startswith("s3://"):
```
[the remark this answers is not in the document]
Comment (yixu34) [reply model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py:459]: +1 about not being cloud-specific (anymore). @squeakymouse is in the process of refactoring this, would be good for her to review.

[the remark this answers is not in the document]
Comment (dmchoiboi) [reply model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py:459]: yeah, this check triggered me a bit as well, but i want to hold off on any big refactors around "cloud agnosticism" until I had a fuller picture on what's needed to achieve it.

File: model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py [hunk 2]
```diff
@@ -770,21 +779,25 @@ async def create_lightllm_bundle(
...
-                    f"Only S3 paths are supported. Given checkpoint path: {checkpoint_path}."
-                )
+        checkpoint_path = (
+            SUPPORTED_MODELS_INFO[model_name].s3_repo if not checkpoint_path else checkpoint_path
+        )
+
+        if not checkpoint_path:
+            raise InvalidRequestException(f"No checkpoint path found for model {model_name}")
+
+        if checkpoint_path.startswith("s3://"):
```
[the remark this answers is not in the document]
Comment (dmchoiboi) [reply model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py]: Cleaned up a bit

File: model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py [hunk 3]
```diff
@@ -835,20 +838,18 @@ async def create_tensorrt_llm_bundle(
...
-                subcommands += self.load_model_files_sub_commands_trt_llm(
-                    checkpoint_path,
-                )
-            else:
-                raise ObjectHasInvalidValueException(
-                    f"Only S3 paths are supported. Given checkpoint path: {checkpoint_path}."
-                )
-        else:
+
+        if not checkpoint_path:
```
Comment (saiatmakuri) [inline model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py:842]: should also use `get_checkpoint_path` here too imo. it's technically net new functionality but should be welcome for tensorrt too

**Review by saiatmakuri — APPROVED** [review]

**Pull request merged**